client/admin/noopClient.go

```diff
@@ -0,0 +1,157 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package admin
+
+import (
+	"context"
+
+	"go.temporal.io/api/serviceerror"
+	"go.temporal.io/server/api/adminservice/v1"
+	"google.golang.org/grpc"
+)
+
+var _ adminservice.AdminServiceClient = (*noopClient)(nil)
+
+var errFakeNotFound = serviceerror.NewNotFound("fake cluster")
+
+type noopClient struct {
+}
+
+// NewNoopClient returns a noop client needed for FakeCluster queue processor
+func NewNoopClient() adminservice.AdminServiceClient {
+	return &noopClient{}
+}
+
+func (n *noopClient) DescribeMutableState(ctx context.Context, in *adminservice.DescribeMutableStateRequest, opts ...grpc.CallOption) (*adminservice.DescribeMutableStateResponse, error) {
+	return nil, errFakeNotFound
+}
+
+func (n *noopClient) DescribeHistoryHost(ctx context.Context, in *adminservice.DescribeHistoryHostRequest, opts ...grpc.CallOption) (*adminservice.DescribeHistoryHostResponse, error) {
+	return nil, errFakeNotFound
+}
+
+func (n *noopClient) GetShard(ctx context.Context, in *adminservice.GetShardRequest, opts ...grpc.CallOption) (*adminservice.GetShardResponse, error) {
+	return nil, errFakeNotFound
+}
+
+func (n *noopClient) CloseShard(ctx context.Context, in *adminservice.CloseShardRequest, opts ...grpc.CallOption) (*adminservice.CloseShardResponse, error) {
+	return nil, errFakeNotFound
+}
+
+func (n *noopClient) ListTransferTasks(ctx context.Context, in *adminservice.ListTransferTasksRequest, opts ...grpc.CallOption) (*adminservice.ListTransferTasksResponse, error) {
+	return nil, errFakeNotFound
+}
+
+func (n *noopClient) ListTimerTasks(ctx context.Context, in *adminservice.ListTimerTasksRequest, opts ...grpc.CallOption) (*adminservice.ListTimerTasksResponse, error) {
+	return nil, errFakeNotFound
+}
+
+func (n *noopClient) ListReplicationTasks(ctx context.Context, in *adminservice.ListReplicationTasksRequest, opts ...grpc.CallOption) (*adminservice.ListReplicationTasksResponse, error) {
+	return nil, errFakeNotFound
+}
+
+func (n *noopClient) ListVisibilityTasks(ctx context.Context, in *adminservice.ListVisibilityTasksRequest, opts ...grpc.CallOption) (*adminservice.ListVisibilityTasksResponse, error) {
+	return nil, errFakeNotFound
+}
+
+func (n *noopClient) RemoveTask(ctx context.Context, in *adminservice.RemoveTaskRequest, opts ...grpc.CallOption) (*adminservice.RemoveTaskResponse, error) {
+	return nil, errFakeNotFound
+}
+
+func (n *noopClient) GetWorkflowExecutionRawHistoryV2(ctx context.Context, in *adminservice.GetWorkflowExecutionRawHistoryV2Request, opts ...grpc.CallOption) (*adminservice.GetWorkflowExecutionRawHistoryV2Response, error) {
+	return nil, errFakeNotFound
```

Comment:
The queue executor will check empty version and return retry error. https://github.com/temporalio/temporal/blob/master/service/history/transferQueueStandbyTaskExecutor.go#L518